### DISCUSSION TITLE
Laravel eloquent bind same parameters to multiple use dont work in mysql

### DISCUSSION THREAD

**Discussion by julianphp** [laravel/framework#39541] [Q&A]
I have a question, as I have a query like this
```
Job::whereRaw('IF (:duedatefrom IS NOT NULL AND :duedateto IS NOT NULL, duedate BETWEEN :duedatefrom AND :duedateto,1)',['duedatefrom' => $duedatefrom,'duedateto' => $duedateto])
    ->get();
```
I have two variables, duedatefrom and duedateto, and I use them in the query two times, then I get the error "**SQLSTATE[HY093]: Invalid parameter number**". 

So what is the correct way to use a variable several times in the query, without having to duplicate the parameters? 
Also, using with postgresql, that same query works perfectly and I can use :duedatefrom and :duedateto as many times as I want.

**Comment by dennisprudlo** [comment]
The keys you specified in the array doesn't matter because the query builder is only interested in the values:
https://github.com/laravel/framework/blob/8e2d72728d6911816a97843ec3341e28c92af120/src/Illuminate/Database/Query/Builder.php#L3262-L3266

Meaning you should pass a simple array with as many values as your query has parameters. The query builder will walk the query query and insert item by item. And because the name of the parameter is irrelevant you can even shorten the query by using `?` instead of `:somename`:

```php
Job::whereRaw(
    'IF (? IS NOT NULL AND ? IS NOT NULL, duedate BETWEEN ? AND ?, 1)',
    [ $duedatefrom, $duedateto, $duedatefrom, $duedateto ]
)->get();
```

**Reply by julianphp** [reply]
Ok, I understand, but I don't understand why it works with postgresql? And I prefer to put a name before ? because if not I think it's more complicated to understand when reading it and it's back to having to replicate the variables.

**Comment by henriquecm** [comment]
According to this stackoverflow answers you can't use the same parameter name more than once in a prepared statement, unless emulation mode is on.

https://stackoverflow.com/a/35375592/2015078

**Reply by julianphp** [reply]
Yes i see, and i dont touch the emulation mode. But it is curious that with postgresql if it works and not with mysql, maybe it changes the mode by itself.
